linux_docker_resources/Dockerfile-RHEL

```diff
@@ -161,14 +160,9 @@ RUN dnf install \
     --refresh -y
 
 # Get and install the RTI web binaries.
-# Connext 5.3.1 for Foxy.
-RUN if [ "$ROS_DISTRO" = "foxy" ]; then \
-       cd /tmp && curl --silent https://s3.amazonaws.com/RTI/Bundles/5.3.1/Evaluation/rti_connext_dds_secure-5.3.1-eval-x64Linux3gcc5.4.0.tar.gz | tar -xz \
-       && tar -xvf /tmp/openssl-1.0.2n-target-x64Linux3gcc5.4.0.tar.gz; \
-    fi
```

Comment:
While I know we didn't officially support RHEL on Foxy, I will point out that the last few releases we did run CI for it, and also package it: https://github.com/ros2/ros2/pull/1334, https://github.com/ros2/ros2/releases/tag/release-foxy-20221021 .  So I kind of think we should keep this around for now, even though we aren't officially supporting it.

That said, this becomes moot in May, when Foxy goes EOL.  So we can definitely remove it then.

Reply:
I've literally never tried Foxy on RHEL, and I'm very pleased to hear that it builds successfully. I honestly didn't think it would work at all, which is why I wanted to drop this.

I'll leave it in then!